us-weather-history/wunderground_parser.py

```diff
@@ -33,34 +33,53 @@ def parse_station(station):
+                if len(weather_data[4]) < 2:
+                    weather_data[4].append(None)
+                    weather_data[4].append(None)
+
+
                 weather_data_units = soup.find(id='historyTable').find_all('td')
 
+
                 try:
-                    actual_mean_temp = weather_data[0].text
-                    actual_max_temp = weather_data[2].text
-                    average_max_temp = weather_data[3].text
-                    record_max_temp = weather_data[4].text
-                    actual_min_temp = weather_data[5].text
-                    average_min_temp = weather_data[6].text
-                    record_min_temp = weather_data[7].text
+                    actual_mean_temp = weather_data[0][0].text
+                    actual_max_temp = weather_data[1][0].text
+                    average_max_temp = weather_data[1][1].text
+                    if weather_data[1][2]:
+                        record_max_temp = weather_data[1][2].text
+                    actual_min_temp = weather_data[2][0].text
+                    average_min_temp = weather_data[2][1].text
+                    record_min_temp = weather_data[2][2].text
                     record_max_temp_year = weather_data_units[
                         9].text.split('(')[-1].strip(')')
                     record_min_temp_year = weather_data_units[
                         13].text.split('(')[-1].strip(')')
 
-                    actual_precipitation = weather_data[9].text
+                    actual_precipitation = weather_data[4][0].text
                     if actual_precipitation == 'T':
-                        actual_precipitation = '0.0'
-                    average_precipitation = weather_data[10].text
-                    record_precipitation = weather_data[11].text
+                        actual_precipitation = '0.0' 
+
+                    #Test whether station collets average or record precipitation data
+                    if weather_data[4][1]:
+                        average_precipitation = weather_data[4][1].text
+                    else: average_precipitation = None
+                    if weather_data[4][2]:
+                        record_precipitation = weather_data[4][2].text
+                    else: record_precipitation = None
 
                     # Verify that the parsed data is valid
                     if (record_max_temp_year == '-1' or record_min_temp_year == '-1' or
                             int(record_max_temp) < max(int(actual_max_temp), int(average_max_temp)) or
                             int(record_min_temp) > min(int(actual_min_temp), int(average_min_temp)) or
-                            float(actual_precipitation) > float(record_precipitation) or
-                            float(average_precipitation) > float(record_precipitation)):
+                            ((record_precipitation is not None or average_precipitation is not None) and (float(actual_precipitation) > float(record_precipitation) or
```

Comment:
Please split the statement following the `and` on the next line, so this line isn't so long.